src/System.Private.CoreLib/src/System/RtType.cs

```diff
@@ -1636,6 +1637,17 @@ internal string GetDefaultMemberName()
 
                 return m_defaultMemberName;
             }
+
+            internal object[] GetEmptyArray()
+            {
+                if (_emptyArray == null)
+                {
+                    MethodInfo method = typeof(Array).GetMethod(nameof(Array.Empty));
```

Comment:
This should not be using reflection to get to `Array.Empty`. Getting to `Array.Empty` this way is expensive and we are not really benefiting from caching done by `Array.Empty` here since we have our own cache.

It should just cache the empty array created by `CreateAttributeArrayHelper` instead.

Reply:
We were worried about allocating a duplicate empty here, but I agree in the scheme of changes here that's a bit silly given we did so for every single call previously - simplified in latest:
```c#
return _emptyArray ?? (_emptyArray = Attribute.CreateAttributeArrayHelper(m_runtimeType, 0));
```